### PR TITLE
Redemption content changes, bugfixes, cancel redemption improvements

### PR DESCRIPTION
**Pull request by mliu**

### CODE REVIEW

File: packages/huma-widget/src/components/Notifi/NotifiContextWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { ChainEnum, getBlockchainConfigFromChain } from '@huma-finance/shared'
+import { NotifiContextProvider } from '@notifi-network/notifi-react'
+import { useWeb3React } from '@web3-react/core'
+import { providers } from 'ethers'
+import React, { useMemo } from 'react'
+
+type Props = {
+  chainId: number | undefined
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  children?: any
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Notifi/NotifiContextWrapper.tsx:11]: Can we add the typings for children?

File: packages/huma-widget/src/components/Notifi/NotifiContextWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    return undefined
+  }, [provider])
+
+  if (account == null || signer == null || chainId == null) {
+    return null
+  }
+
+  let cardId = ''
+  if (chainId === ChainEnum.Celo || chainId === ChainEnum.Alfajores) {
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Notifi/NotifiContextWrapper.tsx:31]: nit: A map might look cleaner

File: packages/huma-widget/src/components/Notifi/NotifiSubscriptionModal.tsx [hunk 3]
```diff
@@ -24,83 +20,48 @@ import { LoadingModal } from '../LoadingModal'
...
-    },
-  )
-
-  return eventTypes
+const validateDefaultTargetGroup = async (
+  frontendClient: NotifiFrontendClient,
+) => {
+  // NOTE: this extra request is necessary as the targetGroupId state in NotifiTargetContext will not be updated constantly right after login
+  const targetGroup = (
+    await frontendClient?.fetchFusionData()
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Notifi/NotifiSubscriptionModal.tsx:32]: nit: I think to have another line to get the fetchFusionData() will look cleaner

File: packages/huma-widget/src/components/Notifi/NotifiSubscriptionModal.tsx [hunk 4]
```diff
@@ -110,43 +71,89 @@ export function NotifiSubscriptionModal({
     setEmailAddress(event.target.value)
   }
 
+  useEffect(() => {
+    async function subscribe() {
+      if (
+        !isInitialized ||
+        !isAuthenticated ||
+        !targetGroupId ||
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Notifi/NotifiSubscriptionModal.tsx:79]: Is the targetGroupId possible to be 0?

**Review by shan-57blocks — APPROVED** [review]

File: packages/huma-widget/src/components/Notifi/NotifiSubscriptionModal.tsx (hunk 4 above)
Comment (mliu) [reply packages/huma-widget/src/components/Notifi/NotifiSubscriptionModal.tsx:79]: No, should be a unique string always

**Pull request merged**